python/ray/data/dataset.py

```diff
@@ -230,10 +231,10 @@ def transform(block: Block) -> Iterable[Block]:
                         "or 'pyarrow', got: {}".format(batch_format))
 
                 applied = fn(view)
-                if isinstance(applied, list) or isinstance(applied, pa.Table):
+                if isinstance(applied, list) \
```

Comment:
Since the if condition is a no-op, this if-else can be simplified to a single error check:
```python
if not (isinstance(applied, list) or
        isinstance(applied, pa.Table) or
        isinstance(applied, pd.core.frame.DataFrame)):
    raise ValueError("The map batches UDF returned the value "
                     f"{applied}, which is not allowed. "
                     "The return type must be either list, "
                     "pandas.DataFrame, or pyarrow.Table")